.travis.yml

```diff
@@ -27,6 +27,10 @@ sudo: required
 before_install:
   - .travis/before_install.sh
 
+before_script:
+  - ./bootstrap
+  - autoreconf --force --install
```

Comment:
+1, that should be / is duplicative to bootstrap. If there's a bug in bootstrap, we should fix it.

Reply:
@samhansen - I think this is the only thing that's blocking the PR from being merged. All else looks fine.

BTW, thanks for the explanation on the `DISTCHECK_CONFIGURE_FLAGS` !

Reply:
The call to autoconf -fi is used to resolve a build failure that manifests as:

```
Making all in lib/support/verhoeff
5906  CXX      VerhoeffTest-VerhoeffTest.o
5907  CXXLD    VerhoeffTest
5908libtool: Version mismatch error.  This is libtool 2.4.6 Debian-2.4.6-0.1, but the
5909libtool: definition of this LT_INIT comes from libtool 2.4.2.
5910libtool: You should recreate aclocal.m4 with macros from libtool 2.4.6 Debian-2.4.6-0.1
```

I think this is versioning mismatch between the libtool being used and the libtool the build was originally based upon.  The autoreconf syncs everything up with the in-tree version.  I suspect that the bootstrap script is not regenerating these file(s) because they exist albeit out of date.

I can think of two options:

1. Identify the stray artifacts and purge them.
2. Add to the bootstrap script this directive.  The local bootstrap script is just a transparent wrapper around the nlbuild-autotools bootstrap script.

Please advise, thanks.

Reply:
If we are removing all artifacts, then libtool is one of them and there should be no conflict.

Reply:
ok, I'll try and track down all the artifacts that get generated and also remove them.  Thanks.

Reply:
I've identified the following additional artifact files (relative to the repository root):

```
aclocal.m4
src/include/BuildConfig.h.in
third_party/nlbuild-autotools/repo/third_party/autoconf/compile
third_party/nlbuild-autotools/repo/third_party/autoconf/config.guess
third_party/nlbuild-autotools/repo/third_party/autoconf/config.sub
third_party/nlbuild-autotools/repo/third_party/autoconf/depcomp
third_party/nlbuild-autotools/repo/third_party/autoconf/install-sh
third_party/nlbuild-autotools/repo/third_party/autoconf/ltmain.sh
third_party/nlbuild-autotools/repo/third_party/autoconf/missing
third_party/nlbuild-autotools/repo/third_party/autoconf/mkinstalldirs
third_party/nlbuild-autotools/repo/third_party/autoconf/py-compile
third_party/nlbuild-autotools/repo/third_party/autoconf/test-driver
```

These files were identified by looking at modified tracked files.  If there are other artifacts that aren't modified by `autoreconf`, I simply don't know what they are.

However, this doesn't resolve the libtool problem.  I think the core problem is that there are currently two build toolchains in play: the hermetic toolchain rooted in nlbuild-autotools, and the distro-provided toolchain on the build host.

Having purged the above files, I think bootstrap is correctly re-initializing everything based on the hermetic toolchain.  This theory is supported by the versioning of libtools stuff pinned at `2.4.2`, which is the hermetic version of libtool.

Later, the host toolchain (or at least parts of the host toolchain) is used for the actual build, and there is a versioning mismatch as on the host libtools is of version `debian-2.4.6-0.1`, which would be a distro-provided package.  The use of `autoreconf -fi` re-syncs everything to the versions of the toolchain present on the host, which is then used to perform the actual build.

I think I remember @robszewczyk mentioning that there is an ongoing story about a fully hermetic build?

However, as it relates to this issue, I think the problem is that, in the case of libtool, we're breaking out of the hermetic build - and that may or may not be expected.

Please advise, thanks.

Reply:
I think removing the `aclocal.m4` should do it.  For completeness, `src/include/BuildConfig.h.in` should also be removed, though clearly that's not the source of differences between `bootstrap` and `autoreconf`.   The files in `third_party/nlbuild-autotools/repo/` should remain as they are.  

I'm wondering whether the `autoreconf` was required because the subprojects have autotools artifacts checked in.  This should not really affect it, but I will test this theory.

Let's not get too distracted by an ongoing story for hermetic builds.  We can land this change and this will be a good change to have, regardless of how we proceed down the road.

Reply:
Nevermind about my comment re: `autoreconf`.  I think the issue is different: `bootstrap` is distinctly different than `autoreconf`, in that `autoreconf` runs everything anchored against the system packages and `bootstrap` runs `libtoolize` anchored against nlbuild-autotools.  I think this is a broad failure in the build system, the `VehoeffTest` is simply the first executable that we encounter.  The same failure occurs against all other executables (e.g. tests)

Reply:
Thanks Rob.  So, if I follow, there may be some issues related to host vs. hermetic build toolchain integration, but it's a separate issue from what this PR addresses?  If that's the case, I'd like to solicit another round of review.  Thanks :)

Reply:
Oh, also, the CI build failure bounces between green/red due to what I believe if flakiness in `happy/tests/standalone/tunnel/test_weave_tunnel_02.py`.

Reply:
Sorry I missed it in Rob's message, I added `aclocal.m4` and `src/include/BuildConfig.h.in` to the set of removed artifacts.  PTAL

Reply:
@samhansen thanks!  I am now back from break; despite my best intentions I have not had a chance to resolve the questions around `bootstrap` vs `autoreconf` w.r.t. `libtoolize`.  I should be able to take a look this evening and provide the update by tomorrow morning

Reply:
ping :)

Reply:
@samhansen, I believe we are pending on another change from @robszewczyk for nlbuild-autotools. @robszewczyk, could you support an argument for prototyping and slipstreaming that change in openweave-core first and then, on successful deployment there, confirm it and merge it in nlbuild-autotools?

Reply:
Thanks, I followed up with @robszewczyk and per his direction ended up including a few more autotools-related changes.  PTAL